tests/feature_privacy_blockvault.py

```diff
@@ -0,0 +1,142 @@
+    
+    reset_statement="DROP TABLE IF EXISTS BlockData; DROP TABLE IF EXISTS SnapshotData; SELECT lo_unlink(l.oid) FROM pg_largeobject_metadata l;"
+    subprocess.check_call(['scripts/postgres_control.sh', 'start', reset_statement])
+    if not dontKill:
+        atexit.register(subprocess.call, ['scripts/postgres_control.sh', 'stop'])
+
+    extraProducerAccounts = Cluster.createAccountKeys(1)
+    vltproducerAccount = extraProducerAccounts[0]
+
+    Print("Launch a cluster")
+    # Producer: node 1, Blockvault access: node 1 and node 2, node 0 standby
+    if cluster.launch(onlyBios=False, pnodes=1, totalNodes=totalNodes,
+                    useBiosBootFile=False, onlySetProds=False, configSecurityGroup=True,
+                    extraNodeosArgs=" --blocks-log-stride 20 --max-retained-block-files 3",
+                    specificExtraNodeosArgs={
+                        1:"--plugin eosio::blockvault_client_plugin --block-vault-backend postgresql://postgres:password@localhost",
+                        2:"--plugin eosio::blockvault_client_plugin --block-vault-backend postgresql://postgres:password@localhost"},
+                    manualProducerNodeConf={ 1: { 'key': vltproducerAccount, 'names': ['vltproducera']}}) is False:
+        Utils.cmdError("launcher")
+        Utils.errorExit("Failed to stand up eos cluster.")
+
+    Print("Validating system accounts after bootstrap")
+    cluster.validateAccounts(None)
+
+    node0 = cluster.getNode(0)
+    node1 = cluster.getNode(1)
+    node2 = cluster.getNode(2)
+
+    securityGroup = cluster.getSecurityGroup()
+
+    time.sleep(10)
+
+    Print("#################################################################################")
+    Print("# Scenario 1: No security group defined#")
+    Print("#################################################################################")
+
+    Print("Wait for blocks produced by vltproducera(node 1) seen by node 0")
+    assert node0.waitForLibToAdvance(), "failed to see blocks produced by vltproducera"
+
+    Print("Wait for blocks produced by vltproducera(node 1) seen by node 2")
+    assert node2.waitForLibToAdvance(), "failed to see blocks produced by vltproducera"
+
+
+    node2.createSnapshot()
+    Print("Wait until the snapshot appears on postgres")
+    assert Utils.waitForTruth(lambda: num_rows_in_table('SnapshotData') != 0), "Node 2 doesn't have access to blockvault" 
+
+    Print("#################################################################################")
+    Print("# Scenario 2: Adding node 0 and node 1 to a security group, check access of node 2 and node 0 to blocks produced by node 1#")
+    Print("#################################################################################")
+
+    Print("Add nodes 0 and 1 to security group")
+    securityGroup.editSecurityGroup(addNodes=[node0, node1])
+    securityGroup.verifySecurityGroup()
+
+    Print("### change postgre password to restrict node 2 access to it ###")
```

Comment:
my understanding of blockvault feature is that it is used to create a cluster with same producer key with backup instances. Hence all of it's instances supposed to have same password and same certificate. So I don't see much logic in the scenario you are testing. Correct me if I missed something.